### PR TITLE
Fix bf16 inference accuracy for mistral, phi3, dbrx

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?

I found that range of models where these changes https://github.com/huggingface/optimum-intel/pull/783 are required are not limited only llama and gemma, the same issue depends on transformers version can be found in phi3, mistral and dbrx.

### CODE REVIEW

File: optimum/exporters/openvino/model_patcher.py [hunk 1]
```diff
@@ -1807,6 +1832,15 @@ def __enter__(self):
             _dbrx_update_causal_mask, self._model.transformer
         )
 
+        # starting from transformers 4.41 issue also observable for calculation sin/cos for rotary_emb
+        patch_rope_sin_cos = is_transformers_version(">=", "4.41.")
```
Comment (echarlaix) [inline optimum/exporters/openvino/model_patcher.py:1836]:
```suggestion
        patch_rope_sin_cos = is_transformers_version(">=", "4.41.0")
```

**Review by echarlaix — APPROVED** [review]
Thanks @eaidova

**Comment by eaidova** [comment]
@echarlaix  could you please merge? These changes are also contributing to support changes in llama model for transformers 4.43

**Comment by echarlaix** [comment]
> @echarlaix could you please merge? These changes are also contributing to support changes in llama model for transformers 4.43

Couple of tests are failing, can you take a look before we can merge ? `test_exporters_cli_int4_with_local_model_and_default_config` and `test_load_model_from_hub_private_with_token` are unrelated

**Comment by eaidova** [comment]
@echarlaix all of them unrelated, gpt-bigcode failed due to issue in optimum patcher (not on our level), optimum-cli failed due to removal config for bloomz from default configs (that leads to changes in group size and config not suitable for testing with small model anymore) - I'll take a look separetly, how we can update this test.

**Comment by echarlaix** [comment]
> @echarlaix all of them unrelated, gpt-bigcode failed due to issue in optimum patcher (not on our level), optimum-cli failed due to removal config for bloomz from default configs (that leads to changes in group size and config not suitable for testing with small model anymore) - I'll take a look separetly, how we can update this test.
> 
> mistral/dbrx - I'm looking

Thanks for fixing ! Will take care of the fix for gpt-bigcode

**Pull request merged**